src/vm/gccover.cpp

```diff
@@ -1371,72 +1479,69 @@ void DoGcStress (PCONTEXT regs, MethodDesc *pMD)
 
     Thread *pThread = GetThread();
 
+    if (!IsGcCoveregeInterruptInstruction((LPVOID)instrPtr))
+    {
+        _ASSERTE(IsOriginalInstruction((LPVOID)instrPtr, gcCover, offset));
+        // Someone beat us to it, just go on running.
+        return;
+    }
+
+    bool atCall;
+    bool afterCallProtect[2] = { 0 };
+
 #if defined(_TARGET_X86_) || defined(_TARGET_AMD64_)
 
     BYTE instrVal = *(BYTE *)instrPtr;
     forceStack[6] = &instrVal;            // This is so I can see it fastchecked
-    
-    if (instrVal != INTERRUPT_INSTR && 
-        instrVal != INTERRUPT_INSTR_CALL && 
-        instrVal != INTERRUPT_INSTR_PROTECT_RET) {
-        _ASSERTE(instrVal == gcCover->savedCode[offset]);  // someone beat us to it.
-        return;       // Someone beat us to it, just go on running
-    }
-
-    bool atCall = (instrVal == INTERRUPT_INSTR_CALL);
-    bool afterCallProtect = (instrVal == INTERRUPT_INSTR_PROTECT_RET);
 
-#elif defined(_TARGET_ARM_)
+    atCall = (instrVal == INTERRUPT_INSTR_CALL);
 
-    WORD instrVal = *(WORD*)instrPtr;
-    forceStack[6] = &instrVal;            // This is so I can see it fastchecked
+    if (instrVal == INTERRUPT_INSTR_PROTECT_BOTH_RET)
+    {
+        afterCallProtect[0] = afterCallProtect[1] = true;
+    }
+    else if (instrVal == INTERRUPT_INSTR_PROTECT_FIRST_RET)
+    {
+        afterCallProtect[0] = true;
+        afterCallProtect[1] = false;
```

Comment:
Since it was already initialized to false, it doesn't seem necessary (or really, more clear) to write `false`.

Reply:
Done.